src/main/java/sg/ncl/MainController.java

```diff
@@ -2042,13 +2042,65 @@ public String experiments(Model model, HttpSession session) throws WebServiceRun
             }
         }
 
+
         model.addAttribute("experimentList", experimentList);
         model.addAttribute("realizationMap", realizationMap);
         model.addAttribute("internetRequestForm", new InternetRequestForm());
 //        System.out.println("Elapsed time to get experiment page:" + (System.currentTimeMillis() - start));
         return EXPERIMENTS;
     }
 
+    @GetMapping(value = "/experiment_profile/{expId}")
+    public String experimentProfile(@PathVariable String expId, Model model, HttpSession session, RedirectAttributes redirectAttributes) {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(properties.getExperiment(expId), HttpMethod.GET, request, String.class);
+
+        log.info("experiment profile: extract experiment");
+        Experiment2 experiment2 = extractExperiment(response.getBody().toString());
+
+        log.info("experiment profile: extract realization");
+        Realization realization = invokeAndExtractRealization(experiment2.getTeamName(), experiment2.getId());
+
+        if (isNotAdminAndNotInTeam(session, realization)) {
+            log.warn("Permission denied to view experiment profile: {} for team: {}", realization.getExperimentName(), experiment2.getTeamName());
+            redirectAttributes.addFlashAttribute(MESSAGE, permissionDeniedMessage);
+            return "redirect:/experiments";
+        }
+
+        User2 experimentOwner = invokeAndExtractUserInfo(experiment2.getUserId());
+
+        /*
+         * get experiment details
+         * returns a json string in the format:
+         * {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)